apps/aecore/lib/aecore/structures/chainstate.ex

```diff
@@ -82,13 +83,13 @@ defmodule Aecore.Structures.Chainstate do
         {valid_chainstate, updated_chainstate} = validate_tx(tx, chainstate_acc, block_height)
 
         if valid_chainstate do
-          {valid_txs_list ++ [tx], updated_chainstate}
+          {[tx | valid_txs_list], updated_chainstate}
         else
           {valid_txs_list, chainstate_acc}
         end
       end)
 
-    valid_txs_list
+    Enum.reverse(valid_txs_list)
```

Comment:
Are you sure we need to keep the same order in this list as before? If not then just returning valid_txs_list is better

Reply:
Yes, it crashes if it's not in the correct order